diff -urN "./Mobius mod/mobius/sql_action.js" ./Mobius/mobius/sql_action.js
--- "./Mobius mod/mobius/sql_action.js"	2020-12-16 19:45:49.698953000 +0100
+++ ./Mobius/mobius/sql_action.js	2020-12-16 19:42:23.786945000 +0100
@@ -311,32 +311,14 @@
                 }
 
                 delete_oldest(connection, obj, count, function (err, results_oldest) { // select oldest
-                    // FIXED Fed4IoT
-                    update_obj={
-                        cni: cni-results_oldest.affectedRows,
-                        cbs: cbs,
-                        st: st,
-                        ri: obj.ri
-                    };
-                    _this.update_cnt_cni(connection, update_obj,function(err2,results2) {
-                        //if (results_oldest.affectedRows == count) {
-                            _this.get_cni_count(connection, obj, function (cni, cbs, st) {
-                                callback(cni, cbs, st);
-                            });
-                        //}
-                        //else {
-                        //    callback(cni, cbs, st);
-                        //}
-                    });
-                    // FIXED Fed4IoT: original code
-                    /*if (results_oldest.affectedRows == count) {
+                    if (results_oldest.affectedRows == count) {
                         _this.get_cni_count(connection, obj, function (cni, cbs, st) {
                             callback(cni, cbs, st);
                         });
                     }
                     else {
                         callback(cni, cbs, st);
-                    }*/
+                    }
                 });
             }
             else {
@@ -1689,34 +1671,7 @@
 };
 
 exports.select_latest_resource = function(connection, parentObj, loop_count, latestObj, callback) {
-    // FIXED Fed4IoT
-
-    /*if(loop_count > 9) {
-        callback('200');
-        return;
-    }*/
-
-    var ty=parseInt(parentObj.ty, 10) + 1
-    var ri=parentObj.ri
-
-    console.time('select_latest ' + ri);
-    var sql = 'select * from (select * from lookup where pi = \'' + ri + '\' and ty = \'' + ty + '\' order by ct desc limit 1)b join ' + responder.typeRsrc[parseInt(ty, 10)] + ' as a on b.ri = a.ri';
-    db.getResult(sql, connection, function (err, results_latest) {
-        console.timeEnd('select_latest ' + ri);
-        if(!err) {
-            if(results_latest.length >= 1) {
-                latestObj.push(results_latest[0]);
-            }
-            callback('200');
-        }
-        else {
-            callback('500-1');
-        }
-    });
-
-    // FIXED Fed4IoT: original code
-
-    /*if(loop_count > 9) {
+    if(loop_count > 9) {
         callback('200');
         return;
     }
@@ -1741,17 +1696,13 @@
         else {
             callback('500-1');
         }
-    });*/
+    });
 };
 
 exports.select_oldest_resource = function(connection, ty, ri, oldestObj, callback) {
     console.time('select_oldest ' + ri);
-    // FIXED Fed4IoT
-    var sql = 'select * from (select * from lookup where pi = \'' + ri + '\' and ty = \'' + ty + '\' order by ct asc limit 1)b join ' + responder.typeRsrc[parseInt(ty, 10)] + ' as a on b.ri = a.ri';
-
-    // FIXED Fed4IoT: original code
-    //var sql = 'select * from (select * from lookup where pi = \'' + ri + '\' and ty = \'' + ty + '\' limit 1)b join ' + responder.typeRsrc[parseInt(ty, 10)] + ' as a on b.ri = a.ri';
-
+    //var sql = util.format('select a.* from (select ri from lookup where (pi = \'%s\') limit 100) b left join lookup as a on b.ri = a.ri where a.ty = \'4\' or a.ty = \'30\' limit 1', ri);
+    var sql = 'select * from (select * from lookup where pi = \'' + ri + '\' and ty = \'' + ty + '\' limit 1)b join ' + responder.typeRsrc[parseInt(ty, 10)] + ' as a on b.ri = a.ri';
     db.getResult(sql, connection, function (err, results_oldest) {
         console.timeEnd('select_oldest ' + ri);
         if(!err) {
